src/transport/SecurePairingSession.cpp

```diff
@@ -427,9 +689,54 @@ CHIP_ERROR SecurePairingSession::HandleCompute_cA(const PacketHeader & header, c
     mDelegate->OnPairingComplete();
 
 exit:
+
+    if (err != CHIP_NO_ERROR)
+    {
+        SendErrorMsg(spake2pErr);
+    }
     return err;
 }
 
+void SecurePairingSession::SendErrorMsg(Spake2pErrorType errorCode)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    System::PacketBufferHandle msg;
+    uint16_t msglen        = sizeof(Spake2pErrorMsg);
+    Spake2pErrorMsg * pMsg = nullptr;
+
+    msg = System::PacketBuffer::NewWithAvailableSize(msglen);
+    VerifyOrExit(!msg.IsNull(), err = CHIP_SYSTEM_ERROR_NO_MEMORY);
+
+    pMsg        = reinterpret_cast<Spake2pErrorMsg *>(msg->Start());
```

Comment:
Again, I am not sure what this overlay struct buys us, and we have to ensure that there is no initial padding in it, etc.